### PR TITLE
:arrow_up: support Python 3.11

### LINKED ISSUE #149: Python 3.11 not supported
Hi, I've tired to run the "working examples" in Python3.11 and get the following error:
image
When running the same example in Python3.9, it works fine.
really cool package! thx for all the support!

### PR DESCRIPTION
**Pull request by jvdd**
Should fix #149 

This PR does the following:
- [x] update `pyproject.toml` and `poetry.lock` to support Python 3.11
- [x] add Python 3.11 to CI-CD matrix
- [x] :bug: update `selenium-wire` dev dependency to v5.x (solves attribute error - https://github.com/wkeeling/selenium-wire/issues/596)
- [x] :bug: fix numpy install issue on Python 3.11
       -> solution: https://python-poetry.org/docs/dependency-specification/#multiple-constraints-dependencies
- [x] :bug: fix install issue for line-profiler and pyarrow on Python 3.11

### CODE REVIEW

**Comment by jonasvdd** [comment]
@jvdd, do the tests run locally?

**Comment by jvdd** [comment]
Nope, exactly the same error! @jonasvdd

File: poetry.lock [hunk 1]
```diff
@@ -246,7 +264,7 @@ toml = ["tomli"]
 
 [[package]]
 name = "cryptography"
-version = "37.0.4"
+version = "38.0.4"
```
Comment (jonasvdd) [inline poetry.lock:267]: This is the cause: https://stackoverflow.com/questions/73859249/attributeerror-module-openssl-ssl-has-no-attribute-sslv3-method

Comment (jonasvdd) [reply poetry.lock:267]: @jvdd

Comment (jvdd) [reply poetry.lock:267]: Isn't this the cause https://github.com/wkeeling/selenium-wire/issues/596

Comment (jvdd) [reply poetry.lock:267]: @jonasvdd

Comment (jvdd) [reply poetry.lock:267]: Updating the `selenium-wire` dev dependency to `5.x` fixes the issue (as was stated in the above selenium-wire issue)

**Review by jonasvdd — APPROVED** [review]

**Comment by jvdd** [comment]
Created pre-release: https://pypi.org/project/plotly-resampler/0.8.4rc1/

If this release allows installation on Python 3.11 we can merge this into main

**Pull request merged**